File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> visitedValues = new HashMap();

        for (int i = 0; i < nums.length; i++) {
            int targetY = target - nums[i];

            Integer visited = visitedValues.get(targetY);
            if (visited != null) {
                return new int[]{visited, i};
            }

            visitedValues.put(nums[i], i);
        }

        return new int[]{};
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        i = 0;
        j = len(nums) - 1
        while i <= j:
            if nums[i] == target:
                return i
            elif nums[j] == target:
                return j
            i += 1
            j -= 1
        return -1